flake.nix

```diff
@@ -0,0 +1,130 @@
+{
+  description = "The PureScript Native Go Compiler Backend";
+
+  inputs = {
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    flake-utils.url = "github:numtide/flake-utils";
+    haskellNix.url = "github:input-output-hk/haskell.nix";
+    nixpkgs.follows = "haskellNix/nixpkgs-unstable";
+  };
+
+  outputs = { self, flake-compat, flake-utils, haskellNix, nixpkgs }:
+    let
+      systems = with flake-utils.lib.system; [
+        x86_64-darwin
+        x86_64-linux
+      ];
+
+      overlay = final: prev: {
+        # This dummy package is required in building the project, but `hsc2hs` is actually provided by GHC.
```

Comment:
It's part of the built GHC env, in any case. I'm not sure if this is the right way of fixing it, but it seems to work.